From 0e81e4689a34e3f413ef403e0d801970d324c5b2 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Fri, 2 Oct 2020 10:06:49 +0100
Subject: [PATCH] overlays: Update display GPIO declarations

The 5.4 kernel changes the way a number of display drivers use GPIOs.
That change has exposed flaws/broken a number of display overlays, so
after a trickle of single-display patches this is a best-guess attempt
to fix the remainder. As none of these changes have been tested on real
displays there is a possibility that this either doesn't fix the
problem or even breaks something that was working - apologies if that
is the case.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/goodix-overlay.dts       | 2 +-
 arch/arm/boot/dts/overlays/hy28a-overlay.dts        | 2 +-
 arch/arm/boot/dts/overlays/hy28b-2017-overlay.dts   | 2 +-
 arch/arm/boot/dts/overlays/media-center-overlay.dts | 6 +++---
 arch/arm/boot/dts/overlays/mz61581-overlay.dts      | 2 +-
 arch/arm/boot/dts/overlays/piscreen-overlay.dts     | 4 ++--
 arch/arm/boot/dts/overlays/piscreen2r-overlay.dts   | 4 ++--
 arch/arm/boot/dts/overlays/sh1106-spi-overlay.dts   | 2 +-
 arch/arm/boot/dts/overlays/ssd1306-spi-overlay.dts  | 2 +-
 arch/arm/boot/dts/overlays/ssd1351-spi-overlay.dts  | 2 +-
 arch/arm/boot/dts/overlays/tinylcd35-overlay.dts    | 4 ++--
 11 files changed, 16 insertions(+), 16 deletions(-)

--- a/arch/arm/boot/dts/overlays/goodix-overlay.dts
+++ b/arch/arm/boot/dts/overlays/goodix-overlay.dts
@@ -31,7 +31,7 @@
 				interrupt-parent = <&gpio>;
 				interrupts = <4 2>; // high-to-low edge triggered
 				irq-gpios = <&gpio 4 0>; // Pin7 on GPIO header
-				reset-gpios = <&gpio 17 0>; // Pin11 on GPIO header
+				reset-gpios = <&gpio 17 1>; // Pin11 on GPIO header
 			};
 		};
 	};
--- a/arch/arm/boot/dts/overlays/hy28a-overlay.dts
+++ b/arch/arm/boot/dts/overlays/hy28a-overlay.dts
@@ -61,7 +61,7 @@
 				fps = <50>;
 				buswidth = <8>;
 				startbyte = <0x70>;
-				reset-gpios = <&gpio 25 0>;
+				reset-gpios = <&gpio 25 1>;
 				led-gpios = <&gpio 18 1>;
 				debug = <0>;
 			};
--- a/arch/arm/boot/dts/overlays/hy28b-2017-overlay.dts
+++ b/arch/arm/boot/dts/overlays/hy28b-2017-overlay.dts
@@ -61,7 +61,7 @@
 				fps = <50>;
 				buswidth = <8>;
 				startbyte = <0x70>;
-				reset-gpios = <&gpio 25 0>;
+				reset-gpios = <&gpio 25 1>;
 				led-gpios = <&gpio 18 1>;
 
 				init = <0x10000e5 0x78F0
--- a/arch/arm/boot/dts/overlays/media-center-overlay.dts
+++ b/arch/arm/boot/dts/overlays/media-center-overlay.dts
@@ -53,9 +53,9 @@
 				bgr;
 				fps = <30>;
 				buswidth = <8>;
-				reset-gpios = <&gpio 23 0>;
+				reset-gpios = <&gpio 23 1>;
 				dc-gpios = <&gpio 24 0>;
-				led-gpios = <&gpio 12 1>;
+				led-gpios = <&gpio 12 0>;
 				debug = <0>;
 			};
 
@@ -66,7 +66,7 @@
 				spi-max-frequency = <2000000>;
 				interrupts = <25 2>; /* high-to-low edge triggered */
 				interrupt-parent = <&gpio>;
-				pendown-gpio = <&gpio 25 0>;
+				pendown-gpio = <&gpio 25 1>;
 				ti,x-plate-ohms = /bits/ 16 <60>;
 				ti,pressure-max = /bits/ 16 <255>;
 			};
--- a/arch/arm/boot/dts/overlays/mz61581-overlay.dts
+++ b/arch/arm/boot/dts/overlays/mz61581-overlay.dts
@@ -65,7 +65,7 @@
 				buswidth = <8>;
 				txbuflen = <32768>;
 
-				reset-gpios = <&gpio 15 0>;
+				reset-gpios = <&gpio 15 1>;
 				dc-gpios = <&gpio 25 0>;
 				led-gpios = <&gpio 18 0>;
 
--- a/arch/arm/boot/dts/overlays/piscreen-overlay.dts
+++ b/arch/arm/boot/dts/overlays/piscreen-overlay.dts
@@ -59,9 +59,9 @@
 				fps = <30>;
 				buswidth = <8>;
 				regwidth = <16>;
-				reset-gpios = <&gpio 25 0>;
+				reset-gpios = <&gpio 25 1>;
 				dc-gpios = <&gpio 24 0>;
-				led-gpios = <&gpio 22 1>;
+				led-gpios = <&gpio 22 0>;
 				debug = <0>;
 
 				init = <0x10000b0 0x00
--- a/arch/arm/boot/dts/overlays/piscreen2r-overlay.dts
+++ b/arch/arm/boot/dts/overlays/piscreen2r-overlay.dts
@@ -59,9 +59,9 @@
 				buswidth = <8>;
 				regwidth = <16>;
 				txbuflen = <32768>;
-				reset-gpios = <&gpio 25 0>;
+				reset-gpios = <&gpio 25 1>;
 				dc-gpios = <&gpio 24 0>;
-				led-gpios = <&gpio 22 1>;
+				led-gpios = <&gpio 22 0>;
 				debug = <0>;
 
                                 init = <0x10000b0 0x00
--- a/arch/arm/boot/dts/overlays/sh1106-spi-overlay.dts
+++ b/arch/arm/boot/dts/overlays/sh1106-spi-overlay.dts
@@ -59,7 +59,7 @@
 				rotate = <0>;
 				fps = <25>;
 				buswidth = <8>;
-				reset-gpios = <&gpio 25 0>;
+				reset-gpios = <&gpio 25 1>;
 				dc-gpios = <&gpio 24 0>;
 				debug = <0>;
 
--- a/arch/arm/boot/dts/overlays/ssd1306-spi-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ssd1306-spi-overlay.dts
@@ -59,7 +59,7 @@
 				rotate = <0>;
 				fps = <25>;
 				buswidth = <8>;
-				reset-gpios = <&gpio 25 0>;
+				reset-gpios = <&gpio 25 1>;
 				dc-gpios = <&gpio 24 0>;
 				debug = <0>;
 
--- a/arch/arm/boot/dts/overlays/ssd1351-spi-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ssd1351-spi-overlay.dts
@@ -59,7 +59,7 @@
 				rotate = <0>;
 				fps = <25>;
 				buswidth = <8>;
-				reset-gpios = <&gpio 25 0>;
+				reset-gpios = <&gpio 25 1>;
 				dc-gpios = <&gpio 24 0>;
 				debug = <0>;
 
--- a/arch/arm/boot/dts/overlays/tinylcd35-overlay.dts
+++ b/arch/arm/boot/dts/overlays/tinylcd35-overlay.dts
@@ -85,9 +85,9 @@
 				fps = <20>;
 				bgr;
 				buswidth = <8>;
-				reset-gpios = <&gpio 25 0>;
+				reset-gpios = <&gpio 25 1>;
 				dc-gpios = <&gpio 24 0>;
-				led-gpios = <&gpio 18 1>;
+				led-gpios = <&gpio 18 0>;
 				debug = <0>;
 
 				init = <0x10000B0 0x80
